Yara module output changes to use a list approach

emmanvg opened the pull request:
Changes output from module from:
``` json
"Yara": {
      "bar_yar:sample_text": {
        "meta": {
          "author": "foo"
        },
        "namespace": "/some/location/yarasigs/Yara-Rules/foo/bar.yar",
        "rule": "sample-text",
        "tags": []
      },
}
```
To the following:
``` json
"Yara": [
     {
        "meta": {
          "author": "foo"
        },
        "namespace": "/some/location/yarasigs/Yara-Rules/foo/bar.yar",
        "rule": "sample-text",
        "tags": []
      },
]
```

emmanvg committed:
use a list approach to for Yara module output

ptcNOP commented:
Can we see what it would take to include that matching strings? Similar to [this](https://github.com/crits/crits_services/blob/master/yara_service/__init__.py#L261) in CRITs.

emmanvg committed:
include strings from match on yara signatures

emmanvg committed:
add logic to include string match data per YARA rule

Added extra config argument to prevent excessive amount of data being created
Log those that surpass the threshold

emmanvg committed:
Merge branch 'dev' into yara-module-output-changes

emmanvg commented:
With the included changes for string matching it would look like the following:
``` json
"Yara": [
     {
        "meta": {
          "author": "foo"
        },
        "namespace": "/some/location/yarasigs/Yara-Rules/foo/bar.yar",
        "rule": "sample-text",
        "strings": [
          {
            "data": "Example",
            "name": "$some_regex",
            "offset": [
              15
            ]
          },
        ],
        "tags": []
      },
]
```

ptcNOP commented:
~Module output looks good. Can we update Report HTML template to show strings?~
